lib/util.c

```diff
@@ -147,3 +151,85 @@ rc_strlcpy(char *dst, char const *src, size_t siz)
 }
 
 #endif
+
+/** Set the network namespace for the current thread (or process - if single threaded).
+ *
+ * @param net_namespace - New network namespace to set.
+ * @param prev_ns_handle - Handle to previous network namespace.
+ *
+ * @return 0 on success, -1 when failure.
+ */
+int rc_set_netns(const char *net_namespace, int *prev_ns_handle)
+{
+    int rc = -1;
+    static const char* crt_nsnet = "/proc/self/ns/net";
+    int sock_ns_fd = -1;
+    char sock_nsnet[NSNET_SZ];
+
+    if (NULL == net_namespace) {
+        rc_log(LOG_ERR, "Namespace not provided");
+        return rc;
+    }
+    if (NULL == prev_ns_handle) {
+        rc_log(LOG_ERR, "NULL NS handle for: %s", net_namespace);
+        return rc;
+    }
+    *prev_ns_handle = -1;
+    snprintf(sock_nsnet, NSNET_SZ, "/var/run/netns/%s", net_namespace);
+    do {
```

Comment:
why do ... while(0)?

Reply:
I think @aravindprasad90 is using `break` to get out of the `do .. while(0)` on error. This is to avoid using `goto`. The same block of code could also have been written as follows using `else` statements instead of `goto` or do `while(0)`.
```
*prev_ns_handle = open(crt_nsnet, O_RDONLY);
if (*prev_ns_handle < 0) {
    rc_log(LOG_ERR, "Cannot open %s errno=%s(%d)", crt_nsnet, strerror(errno), errno);
} 
else {
    sock_ns_fd = open(sock_nsnet, O_RDONLY);
    if (sock_ns_fd < 0) {
        rc_log(LOG_ERR, "Cannot open %s errno=%s(%d)", sock_nsnet, strerror(errno), errno);
    }
    else {
        if (setns(sock_ns_fd, CLONE_NEWNET) < 0) {
            rc_log(LOG_ERR, "'setns' set failed for %s errno=%s(%d)", sock_nsnet,
                   strerror(errno), errno);
        }
        else {
            rc = 0;
        }
    }
}
```